crates/compiler/gen_llvm/src/llvm/build.rs

```diff
@@ -2964,136 +2964,61 @@ fn list_literal<'a, 'ctx>(
     let list_length = elems.len();
     let list_length_intval = env.ptr_int().const_int(list_length as _, false);
 
-    // TODO re-enable, currently causes morphic segfaults because it tries to update
-    // constants in-place...
-    // if element_type.is_int_type() {
-    if false {
+    let is_all_constant = elems.iter().all(|e| e.is_literal());
+
+    if is_all_constant {
         let element_type = element_type.into_int_type();
+
         let element_width = layout_interner.stack_size(element_layout);
-        let size = list_length * element_width as usize;
-        let alignment = layout_interner
+
+        let refcount_slot_bytes = layout_interner
             .alignment_bytes(element_layout)
-            .max(env.target.ptr_width() as u32);
+            .max(env.target.ptr_width() as u32) as usize;
 
-        let mut is_all_constant = true;
-        let zero_elements =
-            (env.target.ptr_width() as u8 as f64 / element_width as f64).ceil() as usize;
+        let refcount_slot_elements =
+            (refcount_slot_bytes as f64 / element_width as f64).ceil() as usize;
```

Comment:
I don't think it makes sense for this to operate in element widths (if this allows compound types). For example `(I128, I128)` has an element width of 32, but the refcount slot bytes will only be 16 in total. So this would incorrectly over estimate the size at one element width. Refcount bytes will only be 8 (most of the time) or 16 (if the list element type has a u128 or i128 in it). So it is unrelated to the number of elements or their size.

**except of course for integers where alignment == width.

Reply:
that's fine, but we still need to reserve the proper padding space given the alignment. that's the purpose of this. if the refcount needs to be placed into the allocated slot, it'll be put at the end of the slot (before the data)